### DISCUSSION TITLE
Shelly HomeKit 2.13.0

### DISCUSSION THREAD

**Discussion by markirb** [mongoose-os-apps/shelly-homekit#1541] [Announcements]
## Changelog

**Important**: This release deliberately breaks the upgrade path from Shelly HomeKit 2.12.X for any **Plus** Devices (those are: +1, +1PM, +i4, +2PM, +RGBWPM, +PlugS). The reason for this is to support direct upgrade from newest stock firmware (1.4.X), that uses a changed partition layout. 
To upgrade current "Shelly HomeKit" devices to this release the valid upgrade path is to:

1. Revert to Stock 1.3.3 (Hit the revert button in webui)
2. Upgrade Stock to current release (1.4.X, this will change partition layout)
3. Upgrade back to Shelly HomeKit Firmware (via the /ota Link)

Your HomeKit configuration/pairing is not lost during this upgrade process. Other/wrong upgrade paths will yield an error during updating, but no harm is done if a wrong firmware is selected.

It should be possible to add the above procedure into the python update tool (flash-shelly.py) in the future for an automated way to upgrade old Shelly HomeKit devices. If anyone can help here this would be much appreciated.

### Features
* Uses new partition layout for Plus Devices and thus can directly be upgraded from Stock FW 1.4.X and reverted back to 1.4.X but no longer from Stock 1.3.3 or Shelly HomeKit 2.12.X
* Shelly Plus AddOn: support up to 5 DS18B20 sesnsors by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1530
* Garage door optionally support swapped sensors by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1527
* Support for Offset Value on Temperature Sensors

### Fixes
* Fix Mini 1 Gen 3 Relais not working immediately after update by @markirb in https://github.com/mongoose-os-apps/shelly-homekit/pull/1521

**Full Changelog**: https://github.com/mongoose-os-apps/shelly-homekit/compare/2.12.2...2.13.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mongoose-os-apps/shelly-homekit/releases/tag/2.13.0'>Shelly HomeKit 2.13.0</a>.</em>

**Comment by felixmueller** [comment]
Hi @markirb,
thanks for the new release!
Just to clarify: I only have Shelly 1 devices (non Plus). Can I just update from file in the Web UI without reverting to stock firmware before because the partition table isn't changed on Shelly 1 devices?

**Reply by markirb** [reply]
Yes correct

**Comment by mappet2011** [comment]
i would love to update, but revert to stock doesnt work, see #1596, any idea? where can i find / download 1.3.3?
